packages/jbrowse-plugin-apollo/src/components/ManageUsers.tsx

```diff
@@ -0,0 +1,217 @@
+import { AbstractSessionModel, AppRootModel } from '@jbrowse/core/util'
+import DeleteIcon from '@mui/icons-material/Delete'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+} from '@mui/material'
+import {
+  DataGrid,
+  GridActionsCellItem,
+  GridCallbackDetails,
+  GridCellEditCommitParams,
+  GridColumns,
+  GridRowId,
+  GridRowParams,
+  GridToolbar,
+  MuiEvent,
+} from '@mui/x-data-grid'
+import { ChangeManager, DeleteUserChange, UserChange } from 'apollo-shared'
+import { getRoot } from 'mobx-state-tree'
+import React, { useCallback, useEffect, useState } from 'react'
+
+import { ApolloInternetAccountModel } from '../ApolloInternetAccount/model'
+
+interface UserResponse {
+  id: number
+  username: string
+  email: string
+  role: string[]
+}
+
+interface ManageUsersProps {
+  session: AbstractSessionModel
+  handleClose(): void
+  changeManager: ChangeManager
+}
+
+export function ManageUsers({
+  session,
+  handleClose,
+  changeManager,
+}: ManageUsersProps) {
+  const { internetAccounts } = getRoot(session) as AppRootModel
+  const apolloInternetAccounts = internetAccounts.filter(
+    (ia) => ia.type === 'ApolloInternetAccount',
+  ) as ApolloInternetAccountModel[]
```

Comment:
Should also probably check here that ia.role has 'admin'

Reply:
fixed the role change ui bug and added the check for admin role. Noticed what looks like a bug in the roles hierarchy being returned in the jwt token from the authentication service:

```
// Loop user's role(s) and add each role + inherited ones to userRolesArray
    for (const userRole of userFound.role) {
      const roles = RoleInheritance[userRole] // Read from role.enum.ts
      roles.forEach((role) => {
        userRoles.add(role)
      })
    }

    const payload = {
      username: userFound.username,
      email: userFound.email,
      roles: Array.from(userRoles),
    }
```
This all makes sense on paper, but I've tested by changing my "admin" account to "readOnly", verified the role has updated in the db, and still the roles array in the jwt token shows all 3 roles (when only [Role.ReadOnly] should be there according to the RoleInheritance). Could this be a caching issue?

Reply:
I think I see what's going on. If you change the role for yourself, until you log out and log in again, you'll still have your old role in the jwt.

Could you update this so that you can't make changes to your own account? Still show the row in the manage users screen, but have the edit/delete functionality disabled. This way we can guarantee that an instance doesn't end up without an admin, and I think it's pretty standard practice in account management.

Reply:
OK I've made the suggested change and it appears to be working as desired now. I added the user id to the jwt payload so that I can retrieve it in the same way I'm doing with the role in ManageUsers component.